### PR TITLE
No allowed failures on PHP 7.2 anymore 

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

**Comment by kukulich** [comment]
Hmm, Travis uses PHP 7.1.1 but the bug in SoapClient is fixed in 7.1.2 :(

**Comment by asgrim** [comment]
@kukulich Hmm, what's the `SoapClient` issue? I just restarted the build but fails again:

```
1) Roave\BetterReflectionTest\SourceLocator\Type\PhpInternalSourceLocatorTest::testAllGeneratedStubsAreInSyncWithInternalReflectionClasses with data set "SoapClient" ('SoapClient')
SoapClient#__setSoapHeaders.soapheaders
Failed asserting that true matches expected false.
/home/travis/build/Roave/BetterReflection/test/unit/SourceLocator/Type/PhpInternalSourceLocatorTest.php:400
/home/travis/build/Roave/BetterReflection/test/unit/SourceLocator/Type/PhpInternalSourceLocatorTest.php:357
/home/travis/build/Roave/BetterReflection/test/unit/SourceLocator/Type/PhpInternalSourceLocatorTest.php:303
/home/travis/build/Roave/BetterReflection/test/unit/SourceLocator/Type/PhpInternalSourceLocatorTest.php:209
```

```
$ php --version
PHP 7.1.11 (cli) (built: Oct 26 2017 16:25:55) ( ZTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.11, Copyright (c) 1999-2017, by Zend Technologies
    with Xdebug v2.5.5, Copyright (c) 2002-2017, by Derick Rethans
```

**Comment by kukulich** [comment]
@asgrim It caused by this: https://github.com/php/php-src/blob/PHP-7.1/NEWS#L170

Travis still runs PHP  7.1.11 unfortunately :(

I can add `if` to the test probably for older PHP.

**Comment by asgrim** [comment]
Ah sorry, your comment suggested it was fixed in 7.1.2 ;) Gotcha. Then yes indeed, we're going to need to hack it to support both versions unfortunately as someone may be trying to reflect `SoapClient` with BR :/

**Comment by kukulich** [comment]
@asgrim Solved :)

File: test/unit/SourceLocator/Type/PhpInternalSourceLocatorTest.php [hunk 1]
```diff
@@ -397,7 +397,12 @@ private function assertSameParameterAttributes(
         }
         //self::assertSame($original->allowsNull(), $stubbed->allowsNull()); @TODO WTF?
         self::assertSame($original->canBePassedByValue(), $stubbed->canBePassedByValue(), $parameterName);
-        self::assertSame($original->isOptional(), $stubbed->isOptional(), $parameterName);
+
+        if (! ($parameterName === 'SoapClient#__setSoapHeaders.soapheaders' && PHP_VERSION_ID < 70112)) {
```
Comment (asgrim) [inline test/unit/SourceLocator/Type/PhpInternalSourceLocatorTest.php]: Disgusting, but works :+1:

**Pull request merged**